src/nvim/api/win_config.c

```diff
@@ -199,7 +199,7 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
     Boolean noautocmd = !enter || fconfig.noautocmd;
     win_set_buf(wp, buf, noautocmd, err);
     if (!fconfig.noautocmd) {
-      apply_autocmds(EVENT_WINNEW, NULL, NULL, false, buf);
+      do_autocmd_with_winid(EVENT_WINNEW, NULL, buf, wp->handle);
```

Comment:
It might be too much noise for this PR, but as a fast-follow would it make sense to add the win parameter to `apply_autocmds`? Because we want to have this field for all events (though initially it can be a "todo").

btw why name it `do_autocmd_with_winid`? The existing functions are:

```
apply_autocmds                     f src/nvim/autocmd.c                                                                                                                  
apply_autocmds_exarg               f src/nvim/autocmd.c                                                                                                                  
apply_autocmds_group               f src/nvim/autocmd.c                                                                                                                  
apply_autocmds_retval              f src/nvim/autocmd.c
```

Reply:
Also `do_autocmd_with_winid` was supposed to be a private function to `window.c`, but it also needed to be called from `nvim_open_win`, so I moved it to `autocmd.c` and made it public.